plugins/follow_me/follow_me_impl.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "follow_me.h"
+#include "mavlink_include.h"
+#include "plugin_impl_base.h"
+#include "device_impl.h"
+#include "timeout_handler.h"
+#include "global_include.h"
+
+/**
+  IMPORTANT NOTE:
+    Macro FOLLOW_ME_TESTING is used to test FollowMe plugin.
+    In real scenario, GCS (DroneCore Application) doesn't provide poistion,
+    but instead, current location of the device is captured by platform-specific Location Framework.
+
+ */
+#ifndef FOLLOW_ME_TESTING
+#define FOLLOW_ME_TESTING
+#endif
+
+namespace dronecore {
+
+class FollowMeImpl : public PluginImplBase
+{
+public:
+    FollowMeImpl();
+    ~FollowMeImpl();
+
+    void init() override;
+    void deinit() override;
+
+    FollowMe::Configuration get_config() const;
+    void set_config(const FollowMe::Configuration &cfg);
+
+    FollowMe::Result start();
+    FollowMe::Result stop();
+private:
+    /* private methods */
+    /*****************************************************/
+    void process_heartbeat(const mavlink_message_t &message);
+    FollowMe::Result to_follow_me_result(MavlinkCommands::Result result) const;
+    void timeout_occurred();
+    void send_gcs_motion_report();
```

Comment:
What does `gcs` mean? I would leave it away, spell it out, or at least add a comment somewhere. It makes code harder to read and understand.

Reply:
Its Ground Control Station. Now I simplified to `send_follow_info()`

Reply:
Cool!